pbxproj/pbxextensions/ProjectFiles.py

```diff
@@ -196,7 +196,7 @@ def get_files_by_name(self, name, parent=None):
 
         files = []
         for file_ref in self.objects.get_objects_in_section(u'PBXFileReference'):
-            if file_ref.name == name and (parent is None or parent.has_child(file_ref.get_id())):
+            if hasattr(file_ref, 'name') and file_ref.name == name and (parent is None or parent.has_child(file_ref.get_id())):
```

Comment:
instead of adding more conditions, and taking into consideration your comment in the PR. I think it's better to create a get_name() method in the PBXFileReference class that does something like this:

```python
    def get_name(self):
        if hasattr(self, u'name'):
            return self.name
        if hasattr(self, u'path'):
            return self.path
        return None
```
and try to use that method instead. The code will remain clean and there is no need to modify whatever is read from the original project.